sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/BatchImpl.java

```diff
@@ -0,0 +1,83 @@
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.Context;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.data.tables.implementation.models.BatchRequestBody;
+import com.azure.data.tables.implementation.models.BatchSubmitBatchResponse;
+import com.azure.data.tables.implementation.models.TableServiceErrorException;
+import reactor.core.publisher.Mono;
+
+/** An instance of this class provides access to all the operations defined in Batch. */
+public final class BatchImpl {
+    /** The proxy service used to perform REST calls. */
+    private final BatchService service;
+
+    /** The service client containing this operation class. */
+    private final AzureTableImpl client;
+
+    /**
+     * Initializes an instance of BatchImpl.
+     *
+     * @param client the instance of the service client containing this operation class.
+     */
+    public BatchImpl(AzureTableImpl client, SerializerAdapter serializerAdapter) {
+        this.service = RestProxy.create(BatchService.class, client.getHttpPipeline(), serializerAdapter);
+        this.client = client;
+    }
+
+    /**
+     * The interface defining all the services for AzureTableBatch to be used by the proxy service to perform REST
+     * calls.
+     */
+    @Host("{url}")
+    @ServiceInterface(name = "AzureTableServices")
+    private interface BatchService {
+        @Post("/$batch")
+        @ExpectedResponses({202})
+        @UnexpectedResponseExceptionType(TableServiceErrorException.class)
+        Mono<BatchSubmitBatchResponse> submitBatch(
+            @HostParam("url") String url,
+            @HeaderParam("Content-Type") String multipartContentType,
+            @HeaderParam("x-ms-version") String version,
+            @HeaderParam("x-ms-client-request-id") String requestId,
+            @HeaderParam("DataServiceVersion") String dataServiceVersion,
+            @BodyParam("multipart/mixed") BatchRequestBody body,
+            Context context);
+    }
+
+    /**
+     * The Batch operation allows multiple API calls to be embedded into a single HTTP request.
+     *
+     * @param body Initial data.
+     * @param requestId Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     *                  analytics logs when storage analytics logging is enabled.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @return a Mono which performs the network request upon subscription.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<BatchSubmitBatchResponse> submitBatchWithRestResponseAsync(
+        BatchRequestBody body,
+        String requestId,
+        Context context
+    ) {
+        final String dataServiceVersion = "3.0";
```

Comment:
Is this the only data service version this is supported in? I thought there was some expandable enum for these.

Reply:
It's the only supported version. The `Impl` types are typically generated by Autorest - if you compare with TablesImpl.java you can see that Autorest generates code with a hard-coded version of "3.0" there as well. In the future we want to replace a lot of the code I wrote here with code that's generated from Autorest, but at present there is some complexity I don't recall that prevents us from being able to describe the batch endpoint correctly in the swagger.